src/sysdir.c

```diff
@@ -85,8 +85,12 @@ static int git_sysdir_guess_template_dirs(git_buf *out)
 
 typedef int (*git_sysdir_guess_cb)(git_buf *out);
 
-static git_buf git_sysdir__dirs[GIT_SYSDIR__MAX] =
-	{ GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT, GIT_BUF_INIT };
+struct git_sysdir__dir {
+	int initialized;
```

Comment:
this looks a littlie bit clumsy, tbh.

Why not simply

  static git_buf *sysdirs 

int init_sysdirs() {
   if (sysdirs != NULL) { return }
   int (**guess)(git_buf *out) = {
      guess_system_dirs, guess_global_dir, ... , NULL };
   sz = 0;
   for (g = guess; *g; g++) { sz ++ }
   sysdirs = malloc(sz*sizeof(git_buf*));

   i = 0;
   for g = guess; *g; g++, i++ { 
     g(&sysdirs[i]);
  }
}